src/tensors/gpu/prod.cpp

```diff
@@ -289,50 +350,68 @@ void ProdBatchedTyped(marian::Tensor C,
   auto strideA = batchA == 1 ? 0 : m * k;
   auto strideB = batchB == 1 ? 0 : n * k;
   auto strideC = n * m;
-  auto batchC = std::max(batchA, batchB);
 
-  std::vector<const T*> aptr;
-  std::vector<const T*> bptr;
-  std::vector<T*> cptr;
+  if(batchA == batchB) {
+    setTensorMode(cublasHandle);
+    CUBLAS_CHECK(cublasGemmBatchedStridedTyped(cublasHandle, 
+                                               compute, 
+                                               opB, 
+                                               opA, 
+                                               n,
+                                               m,
+                                               k,
+                                               &alpha,
+                                               B->data<const T>(), 
+                                               ldb, strideB, 
+                                               A->data<const T>(), 
+                                               lda, strideA, 
+                                               &beta, 
+                                               C->data<T>(), 
+                                               ldc, strideC, 
+                                               batchA));
+    unsetTensorMode(cublasHandle);
+  } else {
+    auto batchC = std::max(batchA, batchB);
```

Comment:
I was confused about what `batchA` etc. mean. Since you are at it, would you mind helping to rename these variables to be more clear? I think it should be `batchDimA`, right? Thanks!

Reply:
Ok I can rename those

Reply:
(Done is recent commits)